src/Controller/Component/PrgComponent.php

```diff
@@ -53,6 +53,35 @@ public function startup()
         return $this->_registry->getController()->redirect($url);
     }
 
+    /**
+     * Populates the $_isSearch view variable based on the current request.
+     *
+     * You need to configure the modelClass config if you are not using the controller's
+     * default modelClass property.
+     *
+     * @return \Cake\Http\Response|null
+     */
+    public function beforeRender()
+    {
+        $controller = $this->_registry->getController();
```

Comment:
You should first check if current action is in list on configured `actions` and bail out early. Current this method will unnecessarily run for all actions.